### PR TITLE
log,importccl,backupccl: add event logs for import and restore jobs

### LINKED ISSUE #57575: backupccl: no event log entries for restore and import
The bulk I/O operations should be reported in the event log. They are not currently.

cc @aaron-crl @piyush-singh

### PR DESCRIPTION
**Pull request by adityamaru**
This change adds a new `CommonJobEventDetails` and two implementations
of the interface in `Import` and `Restore`, to structure the events
emitted during different phases of the job execution.

Both Import/Restore will emit logs at the following stages:
- When the job has completed preparing and is beginning execution.
- When the job has succeeded.
- When the job has started reverting.
- When the job has finished reverting and is in a terminal failed state.

Fixes: #57575

Release note (sql change): IMPORT and RESTORE now emit structured events
to the OPS channel. Refer to the generated documentation for details.

### CODE REVIEW

**Comment by adityamaru** [comment]
I'll add some tests to import and restore to ensure we're emitting at the expected stages of execution but this should be ready for a look. I'm also attaching some screenshots of my local testing.

Screen Shot 2021-03-26 at 9 27 10 AM

Screen Shot 2021-03-26 at 9 40 25 AM

**Review by knz — COMMENTED** [review]
Good approach, I like it.

However, it needs tests.

> Reviewed 16 of 16 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/62658#-:-MWivRHw48n4RhA0ns8T:blbd4m3)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @adityamaru, @dhartunian, and @dt)

___
*[pkg/util/log/eventpb/events.proto, line 62 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/62658#-MWiuuxt7picJ_mb_oPR:-MWiuuxt7picJ_mb_oPS:b-l0rqq0) ([raw file](https://github.com/cockroachdb/cockroach/blob/1e2a7ac4470da2d399a25930deb821c6d1292c3e/pkg/util/log/eventpb/events.proto#L62)):*
> ```protobuf
> 
>   // The type of the job that triggered the event.
>   string job_type = 2 [(gogoproto.customname) = "JobType", (gogoproto.jsontag) = ",omitempty"];
> ```

if this string is known never to contain sensitive information, please mark it as such (see  CommonEventDetails above for an example)
___
*[pkg/util/log/eventpb/events.proto, line 78 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/62658#-MWiv4XIA1FoFiZfxFJT:-MWiv4XIA1FoFiZfxFJU:b-7ropln) ([raw file](https://github.com/cockroachdb/cockroach/blob/1e2a7ac4470da2d399a25930deb821c6d1292c3e/pkg/util/log/eventpb/events.proto#L78)):*
> ```protobuf
>   // The status of the job that triggered the event. This allows the job to
>   // indicate which phase execution it is in when the event is triggered.
>   string status = 6 [(gogoproto.jsontag) = ",omitempty"];
> ```

ditto

**Comment by adityamaru** [comment]
@knz I couldn't find tests for the stuff defined in `eventpb`, maybe I'm just looking in the wrong place? Unless you mean import/restore level tests, which I'm adding soon.

**Comment by knz** [comment]
Another thing: in the release note 

> IMPORT and RESTORE now log to the event log.

There is no such thing as "the event log". You might want to say instead "IMPORT and RESTORE now emit structured events to the OPS channel. Refer to the generated documentation for details."

**Comment by knz** [comment]
>  Unless you mean import/restore level tests, which I'm adding soon.

Yes that is what I mean. You can look at e.g. `TestRotateCerts` in `pkg/security/certs_rotation_test.go` for an example of how to check structured events in a unit test.

**Review by adityamaru — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/62658)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dhartunian, @dt, and @knz)

___
*[pkg/util/log/eventpb/events.proto, line 62 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/62658#-MWiuuxt7picJ_mb_oPR:-MWl9YoxBA2DgkcAJjmD:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/1e2a7ac4470da2d399a25930deb821c6d1292c3e/pkg/util/log/eventpb/events.proto#L62)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

if this string is known never to contain sensitive information, please mark it as such (see  CommonEventDetails above for an example)
</blockquote></details>

Done.
___
*[pkg/util/log/eventpb/events.proto, line 78 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/62658#-MWiv4XIA1FoFiZfxFJT:-MWl9Z5q02zW9yN5TtPd:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/1e2a7ac4470da2d399a25930deb821c6d1292c3e/pkg/util/log/eventpb/events.proto#L78)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

Done.

**Review by knz — APPROVED** [review]
:lgtm: but you need to regenerate some files. 

> Reviewed 11 of 11 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/62658#-:-MWstAuN0OxCd7QRkXlg:b-v37dzp)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @dhartunian and @dt)

**Review by dhartunian — APPROVED** [review]
/pkg/ui changes :lgtm: 

> Reviewed 2 of 16 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/62658#-:-MWyTUH5ELCNsj50sO0Z:b-oktl60)* status: :shipit: complete! 1 of 0 LGTMs obtained (and 1 stale) (waiting on @dt and @knz)

**Comment by adityamaru** [comment]
@dt I'll wait for an LGTM from you before I merge. Made some changes to the import job adding descriptor IDs as well!

File: pkg/ccl/importccl/import_stmt.go
Comment (adityamaru) [reply pkg/ccl/importccl/import_stmt.go]: done.

File: pkg/ccl/backupccl/restore_job.go
Comment (adityamaru) [reply pkg/ccl/backupccl/restore_job.go]: done.

**Comment by adityamaru** [comment]
TFTRs!

bors r+

**Pull request merged**
